### PR TITLE
Add support for reordering items in SimpleFormIterator

### PR DESCRIPTION
**Pull request by djhi**
Supersedes #5835

ra-simpleformiterator-reorder

### CODE REVIEW

File: docs/Inputs.md [hunk 1]
```diff
@@ -86,7 +102,7 @@ Then you can display a text input to edit the author first name as follows:
 ```jsx
 import { BooleanInput } from 'react-admin';
 
-<BooleanInput label="Commentable" source="commentable" />
+<BooleanInput label="Commentable" source="commentable" />;
```
Comment (fzaninotto) [inline docs/Inputs.md:105]: the addition of the final semicolon is wrong (and there are many of these in your commit)

File: docs/Inputs.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: default
-title: "Input Components"
```
Comment (fzaninotto) [inline docs/Inputs.md:3]: please revert the formatting changes in this file, and disable prettier for Markdown in your config. This type of change is a pain to review, may break stuff, and brings no value. Our hand crafted doc is better than what prettier does.

File: packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { MouseEvent } from 'react';
+import { IconButton, IconButtonProps, Tooltip } from '@material-ui/core';
+import { useTranslate } from 'ra-core';
+
+/**
+ * An IconButton with a tooltip which ensures the tooltip is closed on click to avoid ghost tooltips
+ * when the button position changes.
+ */
+export const IconButtonWithTooltip = ({
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx:10]: Not sure this component is useful (see next comment)

File: packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+        handleClose();
+        onClick(event);
+    };
+
+    return (
+        <Tooltip
+            title={translatedLabel}
+            open={open}
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx:36]: I don't understand why you have to handle the open state manually. Tooltip does it on its own, see e.g. https://github.com/marmelab/react-admin/blob/ce116c1059500b8b8758281968f2916bac7b116e/packages/ra-ui-materialui/src/layout/UserMenu.tsx#L86-L95

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx [hunk 5]
```diff
@@ -450,30 +512,11 @@ describe('<SimpleFormIterator />', () => {
         );
 
         expect(queryAllByText('ra.action.remove').length).toBe(0);
+        expect(getByText('Custom Remove Button')).not.toBeNull();
     });
 
-    it('should display the custom add button', () => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx:455]: why did you remove these tests?

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Comment by fzaninotto** [comment]
needs rebase

File: packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx (hunk 4 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/button/IconButtonWithTooltip.tsx:36]: It's explained in the component main comment

File: packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx (hunk 5 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/form/SimpleFormIterator.spec.tsx:455]: I added them back

**Review by fzaninotto — APPROVED** [review]
